### PR TITLE
Split test.cpp into several files

### LINKED ISSUE #55: Split test.cpp into several files
- single-layer test
- graph test
- inference test
- benchmarks
- ...

### PR DESCRIPTION
**Pull request by NeiroYT**
Closes #55

### CODE REVIEW

File: test/single-layer/test_layers.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
+#include <algorithm>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "layers/EWLayer.hpp"
+#include "layers/FCLayer.hpp"
```
Comment (aobolensk) [inline test/single-layer/test_layers.cpp:6]: Split this file into 2: EW and FC layers

**Review by aobolensk — CHANGES_REQUESTED** [review]
Align naming: user "_" as a word splitter in all file names

File: test/benchmarking/test_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,273 @@
+#include <algorithm>
+#include <random>
+#include <thread>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "perf/benchmarking.hpp"
+
+// ==========================
+// Timer tests
```
Comment (aobolensk) [inline test/benchmarking/test_benchmark.cpp:10]: Split timer and accuracy tests

Comment (NeiroYT) [reply test/benchmarking/test_benchmark.cpp:10]: Splitted

File: test/single-layer/test_layers.cpp (hunk 1 above)
Comment (NeiroYT) [reply test/single-layer/test_layers.cpp:6]: Splitted

**Review by aobolensk — APPROVED** [review]

**Review by allnes — APPROVED** [review]

**Pull request merged**